swift/Samples/TabBarContainer/Sources/TabBarScreen.swift

```diff
@@ -0,0 +1,41 @@
+import WorkflowUI
+
+public struct TabBarScreen<Content> {
+
+    public var currentScreen: Content
+    public var barItems: [BarItem]
+    public var selectedIndex: Int
+
+    public init(currentScreen: Content, barItems: [BarItem], selectedIndex: Int) {
+        precondition(barItems.indices.contains(selectedIndex), "selectedIndex \(selectedIndex) is invalid for items \(barItems)")
+
+        self.currentScreen = currentScreen
+        self.barItems = barItems
+        self.selectedIndex = selectedIndex
+    }
+
+}
+
+extension TabBarScreen: Screen where Content: Screen {
+    public func viewControllerDescription(environment: ViewEnvironment) -> ViewControllerDescription {
+        return TabBarScreenContainerViewController.description(for: self, environment: environment)
+    }
+}
+
+public struct BarItem {
+
+    public var title: String
+    public var image: UIImage
+    public var selectedImage: UIImage?
+    public var badge: String?
```

Comment:
I know UITabBarItem uses a `String?` for this, but I’d maybe consider modeling this a little more? Like
```swift
enum Badge {
    case none
    case value(Int)
    case text(String)
}
```

I dunno, might just be based on my aversion to `String?` as a type (i.e. what’s the functional difference between `nil` and `""`?)

Reply:
I chose to keep this the same semantic as what UITabBarItem uses. Although, I like your idea. 

What number formatter would we use when `.value` is used?

Reply:
I’d probably just use `String(value)`